app/src/main/java/com/sample/appconnectsample/ImageCaptureActivity.java

```diff
@@ -0,0 +1,193 @@
+
+    private void dispatchTakePictureIntent() {
+        Intent takePictureIntent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
+        if (takePictureIntent.resolveActivity(getPackageManager()) != null) {
+            startActivityForResult(takePictureIntent, IMAGE_CAPTURE_INTENT);
+        }
+    }
+
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        if (requestCode == IMAGE_CAPTURE_INTENT && resultCode == RESULT_OK) {
+            Bundle extras = data.getExtras();
+            currentImage = (Bitmap) extras.get("data");
+            currentImageView.setImageBitmap(currentImage);
+            submitButton.setEnabled(true);
+        }
+        else if (requestCode == SELECT_IMAGE_INTENT && resultCode == RESULT_OK) {
+            Uri selectedImage = data.getData();
+            try {
+                InputStream imageStream = getContentResolver().openInputStream(selectedImage);
+                currentImage = BitmapFactory.decodeStream(imageStream);
+                currentImageView.setImageBitmap(currentImage);
+                submitButton.setEnabled(true);
+            } catch (FileNotFoundException exception) { }
+        }
+    }
+
+    private void dispatchSelectImageIntent() {
+        Intent photoPickerIntent = new Intent(Intent.ACTION_PICK);
+        photoPickerIntent.setType("image/*");
+        startActivityForResult(photoPickerIntent, SELECT_IMAGE_INTENT);
+    }
+
+    public void doSubmitButton(View source) {
+        submitButton.setEnabled(false);
+        new CollectAndSubmitTask().execute(currentImage);
+    }
+
+    /**
+     * An asynchronous task that collects the responses and submits the form.
+     */
+    private class CollectAndSubmitTask extends AsyncTask<Bitmap,Void,Void> {
+
+        private ProgressDialog progressDialog;
+        private String message;
+        private Exception exception;
+
+        @Override
+        protected void onPreExecute() {
+            progressDialog = new ProgressDialog(ImageCaptureActivity.this, ProgressDialog.STYLE_SPINNER);
+            progressDialog.setMessage(getString(R.string.image_capture_saving_message));
+            progressDialog.setCancelable(false);
+            progressDialog.show();
+        }
+
+        @Override
+        protected Void doInBackground(Bitmap... params) {
+            // *** AppConnect ***
+            // Each secondary thread must create its own datastore instance and
+            // dispose of it when done
+            Datastore datastore = null;
+
+            try {
+                datastore = DatastoreFactory.create();
+                Subject subject = datastore.getSubject(subjectID);
+
+                ByteArrayOutputStream stream = new ByteArrayOutputStream();
+                currentImage.compress(Bitmap.CompressFormat.JPEG, 70, stream);
+                byte[] byteArray = stream.toByteArray();
+
+                // the api call to babbage to actually save the data:
```

Comment:
Nitpick: should we use either `appconnect` or `babbage` exclusively throughout the code comments?